core/src/main/java/com/exadel/aem/toolkit/core/lists/servlets/ListsJsonServlet.java

```diff
@@ -0,0 +1,107 @@
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.gson.Gson;
+import com.day.cq.commons.jcr.JcrConstants;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+
+import com.exadel.aem.toolkit.core.CoreConstants;
+import com.exadel.aem.toolkit.core.lists.ListConstants;
+
+/**
+ * Delivers content of an Exadel Toolbox List as a JSON array
+ * <p><u>Note</u>: This class is not a part of the public API and is subject to change. Do not use it in your own
+ * code</p>
+ */
+@Component(
+    service = Servlet.class,
+    property = {
+        "sling.servlet.resourceTypes=/apps/etoolbox-authoring-kit/lists/components/structure/page-list",
+        "sling.servlet.extensions=json",
+        "sling.servlet.methods=" + HttpConstants.METHOD_GET,
+        "sling.servlet.selectors=list"
+    }
+)
+public class ListsJsonServlet extends SlingSafeMethodsServlet {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ListsJsonServlet.class);
+
+    private static final Gson GSON = new Gson();
+
+    /**
+     * Processes a GET request to deliver the content of an Exadel Toolbox List as a JSON array
+     * @param request  {@code SlingHttpServletRequest} instance
+     * @param response {@code SlingHttpServletResponse} instance
+     * @throws IOException If failed to output to the response stream
+     */
+    @Override
+    protected void doGet(@Nonnull SlingHttpServletRequest request, @Nonnull SlingHttpServletResponse response) throws IOException {
+        PageManager pageManager = request.getResourceResolver().adaptTo(PageManager.class);
+        Page page = pageManager != null ? pageManager.getContainingPage(request.getResource()) : null;
+        if (page == null) {
+            LOG.warn("Resource at {} cannot be adapted to a page", request.getResource().getPath());
+            response.setStatus(HttpStatus.SC_NOT_FOUND);
+            return;
+        }
+
+        Resource listResource = page.getContentResource().getChild("list");
+        if (listResource == null) {
+            LOG.warn("Path {} does not belong to an EToolbox List", request.getResource().getPath());
+            response.setStatus(HttpStatus.SC_NOT_FOUND);
+            return;
+        }
+
+        List<Map<String, Object>> result = new ArrayList<>();
+        for (Resource item : listResource.getChildren()) {
+            if (!ListConstants.LIST_ITEM_RESOURCE_TYPE.equals(item.getResourceType())) {
+                continue;
+            }
+            ValueMap valueMap = item.getValueMap();
+            Map<String, Object> userValues = valueMap
+                .entrySet()
+                .stream()
+                .filter(entry -> JcrConstants.JCR_TITLE.equals(entry.getKey())
+                    || !StringUtils.contains(entry.getKey(), CoreConstants.SEPARATOR_COLON))
```

Comment:
I think this will cover most of cases, as there is a little chance the user will create a cuspom property with a namespace. However, I dont like the fact that we have a slightly different logic for filtering out system poropertis here and in ListItemModel. I think they should work in a same manner